src/Uno.UWP/Globalization/ApplicationLanguages.cs

```diff
@@ -81,10 +82,47 @@ private static void ApplyLanguages()
 			}
 
 			var primaryLanguage = Languages.First();
-			var primaryCulture = new CultureInfo(primaryLanguage);
+			var primaryCulture = CreateCulture(primaryLanguage);
 
 			CultureInfo.CurrentCulture = primaryCulture;
 			CultureInfo.CurrentUICulture = primaryCulture;
 		}
+
+		private static readonly Regex _cultureFormatRegex =
+			new Regex(@"(?<lang>[a-z]{2,8})(?:(?:\-(?<script>[a-zA-Z]+))?\-(?<reg>[A-Z]+))?");
+
+		private static CultureInfo CreateCulture(string cultureId)
+		{
+			try
+			{
+				return new CultureInfo(cultureId);
+			}
+			catch (CultureNotFoundException)
+			{
+				var match = _cultureFormatRegex.Match(cultureId);
```

Comment:
We should create the regex lazily, there's no need to impact the common cases for this. Also, we should be able to detect two `-` to avoid the regex if possible ?